ultravox/training/train.py

```diff
@@ -128,7 +130,17 @@ def train(args: config_base.TrainConfig):
     )
 
     logging.info("Instantiating model...")
-    model = ultravox_model.UltravoxModel(config)
+
+    model_load_context = (
+        accelerate.init_empty_weights()
+        if args.use_fsdp and not is_master
+        else contextlib.nullcontext()
+    )
+    # If we're using FSDP, we can just initialize the model on the main process
+    # and use sync_model_states to distribute the weights to the other processes.
+    # Otherwise we'd be loading the model on every process, which uses too much CPU memory.
+    with model_load_context:
+        model = ultravox_model.UltravoxModel(config)
```

Comment:
This didn't seem necessary for 70B with our current cloud config, but it could easily become a bottleneck.

The issue: The full model gets loaded on RAM per processor and then is moved to GPU. FSDP offers a different approach via the `sync_model_states` option that propagates the weights from the main process.

I've experimented with and without this setting and the training results are exactly the same. I don't have system metrics before W&B starts however to show the expected difference in RAM usage.

Reply:
Good to know.